### PR TITLE
diskutils: refactor partition detection

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Currently, there is an issue where if when we attempt to partition a
virtual disk (/dev/vd*), we make the incorrect choice to append the
"p#" suffix for the partition name, thus failing the partitioning.

Instead of making certain assumptions about the path prefix,
let's actually detect the partition file in /dev and use that
knowledge to inform the rest of the partition initialization

Signed-off-by: Chris Co <chrco@microsoft.com>

###### Change Log  

- Test if specific /dev/<path># file exists. If not, test if /dev/path#p# exists. If not, bail.

###### Does this affect the toolchain?  

NO

###### Test Methodology

Local build

### CODE REVIEW

**Review by niontive — APPROVED** [review]

File: toolkit/tools/imagegen/diskutils/diskutils.go [hunk 1]
```diff
@@ -406,12 +406,28 @@ func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTab
...
+	}
+	if !exists {
+		logger.Log.Warnf("could not find partition in /dev (%s), checking other naming convention", testPartDevPath)
+		testPartDevPath = diskDevPath + "p" + partitionNumberStr
+		exists, err = file.PathExists(testPartDevPath)
+		if err != nil {
+			return
+		}
+		if !exists {
+			logger.Log.Warnf("could not find partition in /dev (%s)", testPartDevPath)
```
Comment (christopherco) [inline toolkit/tools/imagegen/diskutils/diskutils.go:425]: Need to set error here

File: toolkit/tools/imagegen/diskutils/diskutils.go [hunk 2]
```diff
@@ -406,12 +406,28 @@ func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTab
...
+	// There are two primary partition naming conventions:
+	// /dev/sdN<y> style or /dev/loopNp<x> style
+	// Detect the exact one we are using.
+	testPartDevPath := diskDevPath + partitionNumberStr
+	exists, err := file.PathExists(testPartDevPath)
+	if err != nil {
+		return
+	}
+	if !exists {
+		logger.Log.Warnf("could not find partition in /dev (%s), checking other naming convention", testPartDevPath)
```
Comment (christopherco) [inline toolkit/tools/imagegen/diskutils/diskutils.go:418]: Should this be a `Warnf` or `Debugf`?

File: toolkit/tools/imagegen/diskutils/diskutils.go [hunk 3]
```diff
@@ -406,12 +406,35 @@ func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTab
...
+	for _, testPartDevPath := range testPartDevPaths {
+		exists, err = file.PathExists(testPartDevPath)
+		if err != nil {
+			logger.Log.Errorf("Error finding device path (%s)", testPartDevPath)
+			return
+		}
+		if exists {
+			partDevPath = testPartDevPath
+			break
+		} else {
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/diskutils/diskutils.go:427]: Nit: no need for `else` since we're using `break`.

**Review by PawelWMS — APPROVED** [review]
:shipit:

**Pull request merged**